cli/cmd/migration.go

```diff
@@ -0,0 +1,168 @@
+	if err != nil {
+		return err
+	}
+
+	// set new v2 config version and notify our feature event
+	c.CfgVersion = 2
+	c.Event.AddFeatureField("config_version", c.CfgVersion)
+	c.Event.AddFeatureField("org_account", orgInfo.OrgAccount)
+	// NOTE: @afiune this will be a constant pattern below where
+	// we will update settings and notify the feature event
+
+	if orgInfo.OrgAccount {
+		// we only need to update the account/sub-account
+		// if the user has an organizational account
+		c.Log.Debugw("organizational account detected")
+		c.Event.AddFeatureField("org_account_url", orgInfo.OrgAccountURL)
+
+		primaryAccount := strings.ToLower(orgInfo.AccountName())
+
+		// if the user is accessing a sub-account, that is, if the current
+		// account is different from the primary account name, set it as
+		// a what it is, the sub-account
+		if primaryAccount != c.Account {
+			c.Log.Debugw("updating account settings for APIv2",
+				"old_account", c.Account,
+				"new_account", primaryAccount,
+			)
+			c.Subaccount = c.Account
+			c.Account = primaryAccount
+
+			c.Event.AddFeatureField("account", c.Account)
+			c.Event.AddFeatureField("subaccount", c.Subaccount)
+
+			c.Log.Debugw("generating new API client")
+			err = c.NewClient()
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// if the configuration file does not exist, most likely the user
+	// is executing the CLI via env variables or flags, update feature
+	// field and exit migration
+	if !fileExists(viper.ConfigFileUsed()) {
+		c.Log.Debugw("config file not found, skipping profile migration")
+		c.Event.AddFeatureField("config_file", "not_found")
+		return nil
+	}
+
+	c.Log.Debugw("config found, migrating profile", "profile", c.Profile)
+	migratedProfile := lwconfig.Profile{
+		Account:    c.Account,
+		Subaccount: c.Subaccount,
+		ApiKey:     c.KeyID,
+		ApiSecret:  c.Secret,
+		Version:    c.CfgVersion,
+	}
+
+	// create a backup before modifying the user's configuration
+	bkpPath, err := createConfigurationBackup()
+	if err != nil {
+		return err
+	}
+	c.Log.Debugw("configuration backup", "path", bkpPath)
+	c.Event.AddFeatureField("backup_file", path.Base(bkpPath))
+
+	// store the migrated profile
+	err = lwconfig.StoreProfileAt(viper.ConfigFileUsed(), c.Profile, migratedProfile)
+	if err != nil {
+		return errors.Wrap(err, "unable to store migrated profile")
+	}
+
+	c.Log.Debugw("configuration migrated successfully")
+	return nil
+}
+
+func createConfigurationBackup() (string, error) {
+	profiles, err := lwconfig.LoadProfilesFrom(viper.ConfigFileUsed())
+	if err != nil {
+		return "", err
+	}
+
+	cacheDir, err := versionCacheDir()
+	if err != nil {
+		return "", err
+	}
+
+	backupDir := path.Join(cacheDir, ConfigBackupDir)
+	if err := os.MkdirAll(backupDir, 0755); err != nil {
+		return "", err
+	}
+
+	backupCfgPath := path.Join(backupDir,
+		fmt.Sprintf(".lacework.toml.%s.%s.bkp",
+			time.Now().Format("20060102150405"), newID()),
```

Comment:
Is there a const from the time package can be used instead of string "20060102150405"

Reply:
I couldn't find one, do you know if there is one?

At this point I don't think it will be common to create backup files outside a migration so it is not a standard.

Reply:
I had a look there, and as far as i can see there's no const in time package for it. 👍  Leave it as is.